### DISCUSSION TITLE
creating a weakref to pydantic model

### DISCUSSION THREAD

**Discussion by anshdavid** [pydantic/pydantic#2857] [Question] [answered]
Is it possible to create a weakref of a pydantic model ?

```
from pydantic import BaseModel
from uuid import UUID

class JEdgeModel(BaseModel):
    uid: UUID
    startSocket: UUID
    destnSocket: UUID

a = JEdgeModel(uid='abd6fc3f882544f5b75661c92fccbd0d', startSocket='abd6fc3f882544f5b75661c92fccbd0d', destnSocket='abd6fc3f882544f5b75661c92fccbd0d')

wk = weakref.ref(a)
```

I get the following error :
`cannot create weak reference to 'JEdgeModel' object`

Is there a way to achieve the same ?

**Comment by PrettyWood** [comment, marked as answer]
Hi @ironWolf1990 
It's currently not possible directly because `BaseModel` does not have `'__weakref__'` in `__slots__`. Fixing this directly in _pydantic_ is trivial. In the meantime you can add this yourself in a custom `BaseModel`
```py
from pydantic import BaseModel as PydanticBaseModel


class BaseModel(PydanticBaseModel):
    __slots__ = '__weakref__',


class JEdgeModel(BaseModel):
    ...
```
Hope it helps!

**Reply by anshdavid** [reply]
thanks @PrettyWood
